compiler/src/dotty/tools/dotc/classpath/AggregateClassPath.scala

```diff
@@ -124,6 +131,12 @@ case class AggregateClassPath(aggregates: Seq[ClassPath]) extends ClassPath {
             mergedEntries(index) = BinaryAndSourceFilesEntry(existing, entry)
           case (entry: BinaryFileEntry, existing: SourceFileEntry) =>
             mergedEntries(index) = BinaryAndSourceFilesEntry(entry, existing)
+          case (entry: StandaloneTastyFileEntry, _: ClassFileEntry) =>
+            // Here we do not create a TastyFileEntry(entry.file) because the TASTy and the classfile
+            // come from different classpaths. These may not have the same TASTy UUID.
```

Comment:
This comment is confusing. It seems to directly contradict what the commit is trying to do. On the other hand, if I ignore the comment and only read the code, it makes sense. What is that comment really trying to say? Nothing else in this method creates a `TastyFileEntry`; why should I expect that we would do that here?

Reply:
I did not update this documentation correctly after some refactoring. I will update it.

Reply:
Updated

Reply:
The same comment seems to still be here. Perhaps it's meant to say `TastyWithClassFileEntry` instead of `TastyFileEntry`? That would make more sense IIUC.

Reply:
Yes, it should be `TastyWithClassFileEntry`. I updated it.